src/umdsst/Geometry/Geometry.cc

```diff
@@ -38,12 +51,71 @@ namespace umdsst {
 
     atlasFieldSet_.reset(new atlas::FieldSet());
     atlasFieldSet_->add(atlasFunctionSpace_->lonlat());
+
```

Comment:
the land mask will need to be copied as well in the copy constructor here.
I would remove the  above line with `lonlat` and just copy all the fields from `other->atlasFieldSet_` to this class' `atlasFieldSet_`

Reply:
I used atlasFieldSet_-add() method to add (copy) all the fields from "other", hopefully it's ok.